src/onemod/pipeline.py

```diff
@@ -34,12 +38,18 @@ class Pipeline(BaseModel):
     data: Path | None = None
     groupby: set[str] = set()
     _stages: dict[str, Stage] = {}  # set by Pipeline.add_stage
+    _dependencies: dict[str, list[str]] = {}
```

Comment:
Could we simplify some of the validation by using a set instead of a list?

Reply:
Yes. Actually that was my initial thought and implementation, but ultimately thought maybe the target user would be more comfortable with list syntax, at the expense of an extra line of validation. Also was thinking the size of the list of dependencies would not be large enough that we would realize major efficiency improvements.

However, since users won't actually specify dependencies by list going forward, if we do have the concept of stage dependencies under the hood it should probably be set now.